hathor/merged_mining/coordinator.py

```diff
@@ -1222,29 +1302,60 @@ async def update_bitcoin_block(self) -> None:
                 del data_log['transactions']
                 self.log.debug('bitcoin.getblocktemplate response', res=data_log)
                 self.bitcoin_coord_job = BitcoinCoordJob.from_dict(data)
+                assert self.bitcoin_coord_job is not None
                 self.log.debug('new Bitcoin block template', height=self.bitcoin_coord_job.height)
-                await self.update_merged_block()
+                await self.update_merged_block(ParentChain.BITCOIN)
                 if longpoll_id is None:
                     self.log.warn('no longpoll_id received, sleep instead')
                     await asyncio.sleep(self.BITCOIN_UPDATE_INTERVAL)
 
-    def update_bitcoin_submitted(self, height: int) -> None:
+    async def update_digibyte_block(self) -> None:
```

Comment:
I feel we should refactor and move all these specific codes to separate classes, one for each parent chain.

Reply:
I agree it could use a good refactoring. Or even be re-written from scratch. I had very little idea what I would need to generalize while I was adding this support, and even now it isn't too clear. However I'm not sure if it'd be best to do this refactoring right now, at the very least I'd avoid doing this on this PR as not to stretch it for too long.